src/services/api/features/package-actions/lambda-handler.ts

```diff
@@ -0,0 +1,32 @@
+import { type APIGatewayEvent } from "aws-lambda";
+import { AnyZodObject, z } from "zod";
+import { response } from "@/libs/handler";
+
+type LambdaConfig<T extends AnyZodObject, TReturn> = {
+  schema: T;
+  event: APIGatewayEvent;
+  lambda: (data: z.infer<T>) => Promise<TReturn>;
+};
+
+export const handleEvent = async <T extends AnyZodObject, TReturn>({
+  event,
+  schema,
+  lambda,
+}: LambdaConfig<T, TReturn>) => {
+  const body = JSON.parse(event.body ?? "{}");
```

Comment:
By the time we get here, shouldn't the body have been proven not-null/not-undefined by the action router?

Reply:
So the way it's currently setup it's actually the inverse of that. I'm directly passing the event down to this lambda handler from the router

Reply:
AHHH, this is the piece _doing_ that work, not a piece after the work. Got it. 

In that case, keeping with the idea of returning a response object like above, could we check body before we parse so we can differentiate between "missing body" and "malformed body"?